convert/lib/convert.py

```diff
@@ -586,6 +592,42 @@ def get_nukes(stack_name, stack_resources):
             nuke_queue, self.args.remove_security_group_references
         )
 
+    def _print_stacks_status(self, stacks_names: list):
+        for stack_name in stacks_names:
+            stack_status = self.cf.describe_stacks(StackName=stack_name)["Stacks"][0]["StackStatus"]
+            print("Stack:", stack_name, "Status:", stack_status)
+
+    def _delete_stack_wait_for_fail_state(self, stack_name: str, role_arn: str):
+        stack_details = self.cf.describe_stacks(StackName=stack_name)
+        stack_status = stack_details["Stacks"][0]["StackStatus"]
+
+        if stack_status != "DELETE_FAILED":
+            print(f"Deleting Stack: {stack_name} Status: {stack_status}")
+            self.cf.delete_stack(StackName=stack_name, RoleARN=role_arn)
+
+        while (
+            stack_status := self.cf.describe_stacks(StackName=stack_name)["Stacks"][0]["StackStatus"]
+        ) != "DELETE_FAILED":
```

Comment:
Might be worth checking for `in ["DELETE_FAILED", "DELETE_COMPLETED"]`. If either the stack was already deleted and we made it here, or the contents were already wiped out without the stack getting deleted (for whatever reason...) such that even the cloudformation-only IAM role can finish the delete, I think we would endless loop here.